Composer/plugins/runtimes/src/index.ts

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import path from 'path';
+import { promisify } from 'util';
+
+import rimraf from 'rimraf';
+import * as fs from 'fs-extra';
+
+import { copyDir } from './copyDir';
+import { IFileStorage } from './interface';
+
+const exec = promisify(require('child_process').exec);
+
+const removeDirAndFiles = promisify(rimraf);
+
+export default async (composer: any): Promise<void> => {
+  // register the bundled c# runtime used by the local publisher with the eject feature
+  composer.addRuntimeTemplate({
+    key: 'csharp-azurewebapp',
+    name: 'C#',
+    startCommand: 'dotnet run --project azurewebapp',
+    path: path.resolve(__dirname, '../../../../runtime/dotnet'),
+    build: async (runtimePath: string, _project: any) => {
+      composer.log(`BUILD THIS C# PROJECT! at ${runtimePath}...`);
+      composer.log('Run dotnet user-secrets init...');
+      // TODO: capture output of this and store it somewhere useful
+      const { initOut, initErr } = await exec('dotnet user-secrets init --project azurewebapp', {
+        cwd: runtimePath,
+      });
+      if (initErr) {
+        throw new Error(initErr);
+      }
+      composer.log('Run dotnet build...');
+      const { buildOut, buildErr } = await exec('dotnet build', { cwd: runtimePath });
+      if (buildErr) {
+        throw new Error(buildErr);
+      }
+      composer.log('FINISHED BUILDING!');
+    },
+    run: async (project: any, localDisk: IFileStorage) => {
+      composer.log('RUN THIS C# PROJECT!');
```

Comment:
Does this need to be implemented?

Reply:
currently it's just a placeholder, we want to move the spawn process which to host the local runtime into this function in the next step.